### PR TITLE
Add "Cloudflare" extension

### PR DESCRIPTION
**Pull request by Destiner**
## Description

Adds an extension to view personal [Cloudflare](https://www.cloudflare.com) properties (sites, pages, domains, etc).

## Notes

The extension requires a personal API key. I've added instructions to the README page.

## Screencast

https://user-images.githubusercontent.com/4247901/154443554-abe38707-b18d-4a17-8875-b5735da4eb91.mov

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Review by pernielsentikaer — APPROVED** [review]
Hi @Destiner 👋 

Thanks for another great extension, I've tested it out and couldn't find anything to correct, well done! Let's ship it to the next phase 🚢

File: extensions/cloudflare/src/view-pages.tsx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+  useEffect(() => {
+    async function fetchPages() {
+      try {
+        const accounts = await service.listAccounts();
+        if (accounts.length === 0) {
+          setPages([]);
+          setLoading(false);
+        }
+        const account = accounts[0];
```
Comment (mathieudutour) [inline extensions/cloudflare/src/view-pages.tsx:33]: Out of curiosity, could there be ore than 1 account? and in that case would it make sense to list the pages for all of them (maybe using `List.Section`)?

File: extensions/cloudflare/src/view-sites.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+function SiteView(props: SiteProps) {
+  const { accountId, id } = props;
+
+  const [site, setSite] = useState<Zone | null>(null);
+  const [isLoading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchSite() {
+      const site = await service.getZone(id);
```
Comment (mathieudutour) [inline extensions/cloudflare/src/view-sites.tsx:88]: do we need to `try/catch` here as well?

File: extensions/cloudflare/src/view-sites.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+function DnsRecordView(props: DnsRecordProps) {
+  const { siteId } = props;
+
+  const [records, setRecords] = useState<DnsRecord[]>([]);
+  const [isLoading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchRecords() {
+      const records = await service.listDnsRecords(siteId);
```
Comment (mathieudutour) [inline extensions/cloudflare/src/view-sites.tsx:149]: `try/catch`?

File: extensions/cloudflare/src/view-pages.tsx (hunk 1 above)
Comment (Destiner) [reply extensions/cloudflare/src/view-pages.tsx:33]: I think that might be the case, will update to handle it.

File: extensions/cloudflare/src/view-sites.tsx (hunk 2 above)
Comment (Destiner) [reply extensions/cloudflare/src/view-sites.tsx:88]: Sounds good 👍

Comment (Destiner) [reply extensions/cloudflare/src/view-sites.tsx]: Fixed in 570669becea6763affe62d25b16e1b4d27798352.

File: extensions/cloudflare/src/view-sites.tsx (hunk 3 above)
Comment (Destiner) [reply extensions/cloudflare/src/view-sites.tsx]: Fixed in 570669becea6763affe62d25b16e1b4d27798352.

File: extensions/cloudflare/src/view-pages.tsx (hunk 1 above)
Comment (Destiner) [reply extensions/cloudflare/src/view-pages.tsx]: Fixed in c99bb3fabb990ecea3fbd556e9d6a44a0473a66d.

**Review by mathieudutour — APPROVED** [review]
:100:

**Pull request merged**